homeassistant/components/apcupsd/config_flow.py

```diff
@@ -0,0 +1,161 @@
+        return OptionsFlowHandler(config_entry)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            # Test if connection to the host is ok and get the current status for later
+            # configuration.
+            data_service = APCUPSdData(user_input[CONF_HOST], user_input[CONF_PORT])
+            try:
+                status: dict[str, Any] | None = await self.hass.async_add_executor_job(
+                    lambda: data_service.status
+                )
+                if status is None:
+                    return self.async_abort(reason="no_status")
+
+                # We _try_ to use the serial number of the UPS as the unique id since
+                # this field is not guaranteed to exist on all APC UPS models.
+                if "SERIALNO" in status:
+                    await self.async_set_unique_id(status["SERIALNO"])
+                    self._abort_if_unique_id_configured()
+
+                # If we import from YAML, user_input may contain CONF_RESOURCES which
+                # should be set as options.
+                # Note that it may contain unavailable resources, so we need to filter
+                # them out and log warnings for them.
+                options = None
+                if self.source == SOURCE_IMPORT and CONF_RESOURCES in user_input:
+                    available_resources = set(data_service.status.keys())
+                    resources = []
+                    for resource in user_input[CONF_RESOURCES]:
+                        if resource not in available_resources:
+                            _LOGGER.warning(
+                                "Sensor type: %s does not appear in the APCUPSd status output and will be ignored",
+                                resource,
+                            )
+                            continue
+                        resources.append(resource)
+
+                    options = {CONF_RESOURCES: resources}
+
+                return self.async_create_entry(
+                    # Since the MODEL field is not always available on all models, we
+                    # try to set a friendly name for the integration, otherwise default
+                    # to "APC UPS".
+                    title=status.get("MODEL", "APC UPS"),
+                    description="APCUPSd",
+                    data={
+                        CONF_HOST: user_input[CONF_HOST],
+                        CONF_PORT: user_input[CONF_PORT],
+                    },
+                    options=options,
+                )
+            except OSError:
+                errors["base"] = "cannot_connect"
+
+        schema = vol.Schema(
+            {
+                vol.Required(CONF_HOST, default="localhost"): cv.string,
+                vol.Required(CONF_PORT, default=3551): cv.port,
```

Comment:
Use a selector instead for the port.

Example:
https://github.com/home-assistant/core/blob/cd1227d8b9c4ae677f1218aeabc9dea497393db9/homeassistant/components/mysensors/config_flow.py#L48-L55

Reply:
Done!